src/hooks/useQuizTrackingAndCbEvents.ts

```diff
@@ -0,0 +1,183 @@
+/* eslint-disable max-params */
+/* eslint-disable @typescript-eslint/naming-convention */
+import ConstructorIOClient from '@constructor-io/constructorio-client-javascript';
+import { GetBrowseResultsResponseData } from '@constructor-io/constructorio-client-javascript/lib/types';
+import { useCallback, useEffect } from 'react';
+import { ActionAnswerQuestion, QuestionTypes } from '../components/CioQuiz/actions';
+import { QuizAPIReducerState } from '../components/CioQuiz/quizApiReducer';
+import { ResultsPageOptions } from '../components/Results/Results';
+import { isFunction } from '../utils';
+
+const useQuizTrackingAndCbEvents = (
```

Comment:
there is a lot going on in here. would love to explore options for breaking down this complexity for maintainability purposes